client/me/next-steps/index.jsx

```diff
@@ -130,91 +101,84 @@ const NextSteps = createReactClass( {
 				</div>
 			);
 		}
-	},
+	}
 
-	newestSite: function() {
-		return sortBy( this.props.sites.get(), property( 'ID' ) ).pop();
-	},
+	renderOutroMessage() {
+		const { isWelcome, newestSite, translate } = this.props;
 
-	outroMessage: function() {
-		var site, dismissLink;
-
-		if ( this.props.isWelcome ) {
-			site = this.newestSite();
-			dismissLink = '/stats/insights/' + ( site ? site.slug : '' );
+		if ( isWelcome ) {
+			const dismissLink = '/stats/insights/' + ( newestSite ? newestSite.slug : '' );
 
 			return (
 				<div className="next-steps__outro">
 					<p>
-						{ this.props.translate(
+						{ translate(
 							'If you want you can {{a}}skip these steps{{/a}}. You can come back to this page any time.',
 							{
 								components: {
-									a: <a href={ dismissLink } onClick={ this.dismissLinkRecordEvent } />,
+									a: <a href={ dismissLink } onClick={ this.recordDismissLinkClick } />,
 								},
 							}
 						) }
 					</p>
 				</div>
 			);
 		}
-	},
-
-	userHasPurchasedAPlan: function() {
-		return this.props.sites.get().some( function( site ) {
-			return productsValues.isPlan( site.plan );
-		} );
-	},
+	}
 
-	renderSteps: function() {
-		var site = this.newestSite(),
-			sequence = steps.defaultSequence;
+	renderSteps() {
+		const { hasPlan, isWelcome, newestSite } = this.props;
+		let sequence = steps.defaultSequence;
 
-		if ( this.userHasPurchasedAPlan() ) {
+		if ( hasPlan ) {
 			sequence = steps.hasPlanSequence;
 		}
 
 		return (
 			<div className="next-steps__steps">
-				{ sequence.map(
-					function( stepName, index ) {
-						var step = steps.definitions( site )[ stepName ];
-						return (
-							<NextStepsBox
-								key={ stepName }
-								stepName={ stepName }
-								step={ step }
-								primary={ index === 0 }
-								isWelcome={ this.props.isWelcome }
-							/>
-						);
-					}.bind( this )
-				) }
+				{ sequence.map( ( stepName, index ) => {
+					const step = steps.definitions( newestSite )[ stepName ];
+
+					return (
+						<NextStepsBox
+							key={ stepName }
+							stepName={ stepName }
+							step={ step }
+							primary={ index === 0 }
+							isWelcome={ isWelcome }
+						/>
+					);
+				} ) }
 			</div>
 		);
-	},
-
-	render: function() {
-		var classes = 'main main-column next-steps';
+	}
 
-		this.setUserStateFromTrophiesData();
-
-		if ( this.props.isWelcome ) {
-			classes += ' is-single-page';
-		}
+	render() {
+		const classes = classNames( 'main', 'main-column', 'next-steps', {
```

Comment:
Should we use the `Main` component?

Reply:
I'll leave that for a future PR, we have a ton of occurrences similar to `<div className="main main-column next-steps">` which probably precede the creation of `<Main />`.